Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -376,10 +376,14 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
     func SELdidSelectEngine(sender: UIButton) {
         // The UIButtons are the same cardinality and order as the array of quick search engines.
         // Subtract 1 from index to account for magnifying glass accessory.
-        if let index = searchEngineScrollViewContent.subviews.indexOf(sender),
-           let url = quickSearchEngines[index - 1].searchURLForQuery(searchQuery) {
-            searchDelegate?.searchViewController(self, didSelectURL: url)
-        }
+        guard let index = searchEngineScrollViewContent.subviews.indexOf(sender) else { return }
+
+        let engine = quickSearchEngines[index - 1]
+
+        guard let url = engine.searchURLForQuery(searchQuery) else { return }
+
+        Telemetry.recordEvent(SearchTelemetryFactory.makeEvent(engineID: engine.id, source: .QuickSearch))
```

Comment:
Neither of those `guard`s should ever fail (I should probably add `assertionFailures()` in them), so it shouldn't really matter either way.